### PR TITLE
use the md5 crate

### LINKED ISSUE #117: Replace Emacs' MD5 implementation with a Rust crate

### PR DESCRIPTION
**Pull request by db48x**
Fixes #117

### CODE REVIEW

File: lib/md5.h [hunk 1]
```diff
@@ -49,85 +45,19 @@
...
-   always in little endian byte order, so that a byte-wise output yields
-   to the wanted ASCII representation of the message digest.  */
-extern void *__md5_read_ctx (const struct md5_ctx *ctx, void *resbuf) __THROW;
-
-
 /* Compute MD5 message digest for LEN bytes beginning at BUFFER.  The
    result is always in little endian byte order, so that a byte-wise
    output yields to the wanted ASCII representation of the message
    digest.  */
 extern void *__md5_buffer (const char *buffer, size_t len,
```
Comment (DavidDeSimone) [inline lib/md5.h]: Do we still need this function declaration here? Can this be removed like the others?

File: lib/gnulib.mk.in [hunk 2]
```diff
@@ -1091,9 +1091,7 @@ endif
 ## begin gnulib module crypto/md5
 ifeq (,$(OMIT_GNULIB_MODULE_crypto/md5))
 
-libgnu_a_SOURCES += md5.c
-
-EXTRA_DIST += gl_openssl.h md5.h
+EXTRA_DIST += md5.h
```
Comment (DavidDeSimone) [inline lib/gnulib.mk.in]: This file is autogen'd, @birkenfeld should we be regenerating this via gnulib-tool, are we ok with just raw editing this file for the remacs project?

File: lib/md5.h (hunk 1 above)
Comment (db48x) [inline lib/md5.h]: Yes, this declares the md5_buffer function that the rest of the emacs code will call.

Comment (birkenfeld) [inline lib/md5.h]: It's currently the same for the other crypto function, so that's fine.

File: lib/gnulib.mk.in (hunk 2 above)
Comment (birkenfeld) [inline lib/gnulib.mk.in]: I haven't looked into gnulib-tool or admin/merge-gnulib or similar, so far I've edited all relevant files by hand. But if there's a better way we can investigate it later.

**Comment by birkenfeld** [comment]
Closing in favor of #260.

**Pull request closed without merging**